Simplify command to get algo/keytype

cuonglm opened the pull request:
By using just one sed command instead of chaining grep | sed. Save us
one process per call, and also still working if the server returns more
than two lines contains "sign_and_send_pubkey:".

cuonglm commented:
Kindly ping @bk2204 @tgummerer

tgummerer commented:
@cuonglm thanks for the PR!  To be honest this goes above my level of knowledge of `sed`.  Overall I'm finding the end result harder to parse, and I don't think saving one process per call is worth it, since this is a one-off script, rather than something I'd expect to be run continuously anywhere.  

> also still working if the server returns more than two lines contains "sign_and_send_pubkey:".

This might be able to convince me.  I don't think this should typically be the case, but maybe there's a simpler way to achieve the same goal without resorting to more complicated sed scripts?

cuonglm commented:
@tgummerer I can separate the sed script and added the comment, if you want to clarify. Would you point out what part make you confuse? Honestly, it's just the same sed script you are using, just without the `grep`:

 - For any line that matched the regex, don't `!d`elete it.
 - For that matched line, perform `s`ubstitution  (`//` matched previous regex).

The fact that I changed `\([a-z0-9-]*\)` to `\([a-z0-9-][a-z0-9-]*\)` to strictly match the `\([a-z0-9-]+\)` pattern, since when the `+` operator is only available in ERE and not all sed versions support that.

For the current `grep -v`, if it returns more than 1 lines, the `KEYTYPE` will contains weird result. While with the `sed` version, only strictly matched line will result in `KEYTYPE`, which I think be better.

cuonglm commented:
@tgummerer I updated with simpler sed script, and keeping the regex matches the current version. PTAL!

tgummerer commented:
@congulm Thanks for the explanation here!

> The fact that I changed `\([a-z0-9-]*\)` to `\([a-z0-9-][a-z0-9-]*\)` to strictly match the `\([a-z0-9-]+\)` pattern, since when the `+` operator is only available in ERE and not all sed versions support that.

Right that makes sense, but ideally would be in a separate commit, describing the change.  Since it wasn't described anywhere it wasn't obvious to me why this would have been useful.



> * For any line that matched the regex, don't `!d`elete it.
> 
> * For that matched line, perform `s`ubstitution  (`//` matched previous regex).

Yeah this seems not too complicated, *but* is more `sed` than I ever learned :) not sure about others, but I feel like less of the terse sed syntax is better.

I think the newest improvements are good, thank you!
